node-api/handlers/utils/id.go

```diff
@@ -112,13 +112,14 @@ func IsTimestampIDPrefix(timestampID string) bool {
 }
 
 // slotFromStateID returns a slot number from the given state ID.
-// TODO: This pattern does not allow us to query block 0. Genesis points to block 1.
+// In case of genesis, we will return 1 as the slot number. This is to ensure
+// validator API endpoints return a valid slot number for genesis.
```

Comment:
Here keeping the genesis slot as 1 (as it was earlier, just removed the Genesis constant) so that the APIs that use this function can work. As in if we return 0, we will get the latest slot.

Will have to find a way to handle the special genesis case for all these APIs, as part of separate PR as it is not related to Block header.